src/Main.elm

```diff
@@ -1,16 +1,411 @@
+        else
+            li [ class "tag is-primary" ]
+                [ a
+                    [ onClick (SelectBreadcrumbItem folder)
+                    ]
+                    [ text <| folder.name ]
+                ]
+
+
+
+-- VIEW content - subfolders
+
+
+viewFolderContentSubFolders : Maybe AudienceFolder -> List AudienceFolder -> Html Msg
+viewFolderContentSubFolders selectedFolder audienceFolders =
+    case selectedFolder of
+        Just selectedFolder_unwrap ->
+            ul [ id "contentPanel folderList" ]
+                (audienceFolders
+                    |> List.filterMap
+                        (viewFolderContentSubFolderItem
+                            selectedFolder_unwrap
+                        )
+                )
+
+        Nothing ->
+            div [ class "notification is-warning" ] [ text "no select folder - no subfolders" ]
+
+
+viewFolderContentSubFolderItem : AudienceFolder -> AudienceFolder -> Maybe (Html Msg)
+viewFolderContentSubFolderItem selectedFolder folder =
+    let
+        isOpened =
+            isOpenFolder selectedFolder folder
+    in
+        case folder.parent of
+            Just parentId ->
+                if (Debug.log "parent id" parentId) == (Debug.log "selectedFolder.id" selectedFolder.id) then
+                    Just <|
+                        li
+                            [ class "button is-info" ]
+                            [ a
+                                [ onClick (SelectSubDirectory folder)
+                                ]
+                                [ Icon.folder
+                                , text folder.name
+                                ]
+                            ]
+                else
+                    Nothing
+
+            Nothing ->
+                Nothing
+
+
+
+-- VIEW content - audiences
+
+
+viewFolderContentAudiences : Maybe AudienceFolder -> List Audience -> Html Msg
+viewFolderContentAudiences selectedFolder audiences =
+    case selectedFolder of
+        Just selectedFolder_unwrap ->
+            ul [ id "contentPanel audienceList" ]
+                (audiences
+                    |> List.filterMap
+                        (viewContentAudienceItem selectedFolder_unwrap)
+                )
+
+        Nothing ->
+            div [ class "notification is-warning" ] [ text "no select folder - no audiences" ]
+
+
+viewContentAudienceItem : AudienceFolder -> Audience -> Maybe (Html Msg)
+viewContentAudienceItem selectedFolder audience =
+    case audience.folder of
+        Just folderId ->
+            if folderId == selectedFolder.id then
+                Just (div [] [ text audience.name ])
+            else
+                Nothing
+
+        Nothing ->
+            Nothing
+
+
+
+-- UTILS
+
+
+isOpenFolder : AudienceFolder -> AudienceFolder -> Bool
+isOpenFolder selectedFolder folder =
+    if folder.id == selectedFolder.id then
+        True
+    else
+        False
```

Comment:
there is quite some redundant code.

```
folder.id == selectedFolder.id
```

will do the same trick (since it's expression which return Boolean)